### PR TITLE
Optionally cache credentials to disk

### PR DESCRIPTION
**Pull request by oremj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,12 @@ Assuming account1 is the account alias in ~/.aws/accounts and root is the role t
 aws-assume-role account1 root
 ```
 
+## Options
+
+`AWS_ASSUME_ROLE_CACHE_TO_FILE`
+
+If `AWS_ASSUME_ROLE_CACHE_CREDENTIALS_TO_FILE=true` is set, the master session credentials will be written to file and will be reused for multiple terminal sessions.
```
Comment (jvehent) [inline README.md]: should say `is set to "true"`

File: aws-assume-role [hunk 2]
```diff
@@ -7,6 +7,9 @@ MFA_DEVICE=$(cat "${MFA_DEVICE_FILE}")
 AWS_ASSUME_ROLE_SESSION_TIMEOUT=${AWS_ASSUME_ROLE_SESSION_TIMEOUT:-7200}
 AWS_ASSUME_ROLE_DEFAULT_REGION=${AWS_ASSUME_ROLE_DEFAULT_REGION:-us-west-2}
 
+AWS_ASSUME_ROLE_CACHE_TO_FILE=${AWS_ASSUME_ROLE_CACHE_TO_FILE:-false}
+AWS_ASSUME_ROLE_CACHE_FILE=${AWS_ASSUME_ROLE_CACHE_FILE:-~/.aws/.aws-assume-role-master-creds}
```
Comment (jvehent) [inline aws-assume-role]: I think it should default to a file in /tmp to get wiped across reboots

**Comment by jvehent** [comment]
I'm not concerned about leaking credentials because 1) they are temporary and 2) the permissions are the same that are needed to read memory on the host.

I think the greater risk comes from people who might run commands against production while thinking they are on a different profile. I'm not sure how we can help prevent that. I use a modified PS2 bash prompt, but it's hardly something we can expect from everyone.

File: aws-assume-role (hunk 2 above)
Comment (oremj) [reply aws-assume-role]: How would we know what the file is across terminal sessions?

**Comment by oremj** [comment]
If the user is using zsh it automatically sets up the prompt stuff. This is what mine looks like: 
image

**Comment by jvehent** [comment]
ya fancy hipster

File: aws-assume-role (hunk 2 above)
Comment (jvehent) [reply aws-assume-role]: It can be a predefined name, I'm just saying we should put it in /tmp instead of the home directory.

Comment (oremj) [reply aws-assume-role]: ah, gotcha

**Review by jvehent — APPROVED** [review]
r+

**Pull request merged**